executor/instruction/mod.rs

```diff
@@ -67,6 +71,8 @@ mod sub_reverse_executor;
 pub(crate) mod tuple;
 mod type_list_executor;
 
+pub(crate) const TYPES_EMPTY: Vec<Type> = Vec::new();
```

Comment:
Why is this necessary? Surely you can just use `Vec::new()` inline?

Reply:
Easier to read IMO